### PR TITLE
Fix loss inflation under gradient accumulation for ForImageClassifica…

### PR DESCRIPTION
**Pull request by Ifeoluwah**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48207) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48207)

Part of #47688 

## The bug

`Trainer.training_step` decides whether to divide the loss by
`current_gradient_accumulation_steps` based on
`self.model_accepts_loss_kwargs`, which is set from a signature check: if a
model's `forward()` declares `**kwargs`, the Trainer assumes the model will
normalize the loss itself using `num_items_in_batch`, and skips its own
normalization.

Several `*ForImageClassification` / `*ForVideoClassification` heads declare
`**kwargs` in `forward()` but never forward it (or `num_items_in_batch`
specifically) into `self.loss_function(...)`. That satisfies the Trainer's
signature check without actually doing anything with it, so **neither**
normalization happens — the reported loss, and its gradient, end up inflated
by roughly the gradient-accumulation step count.

`SwinForImageClassification` already does this correctly
(`self.loss_function(labels, logits, self.config, **kwargs)`).
`Swinv2ForImageClassification`, right next to it, drops the `**kwargs`
(`self.loss_function(labels, logits, self.config)`) — same codebase, same
loss function, silently different behavior under gradient accumulation.

### Repro

```python
import torch
from transformers import SwinConfig, SwinForImageClassification, Swinv2Config, Swinv2ForImageClassification

kwargs = dict(image_size=32, patch_size=2, embed_dim=8, depths=[1, 1],
              num_heads=[1, 1], window_size=2, num_labels=3)
torch.manual_seed(0)
pixel_values, labels = torch.randn(4, 3, 32, 32), torch.tensor([0, 1, 2, 0])

for name, cls, cfg in [("Swin", SwinForImageClassification, SwinConfig),
                        ("Swinv2", Swinv2ForImageClassification, Swinv2Config)]:
    torch.manual_seed(0)
    model = cls(cfg(**kwargs)).eval()
    with torch.no_grad():
        l4 = model(pixel_values=pixel_values, labels=labels, num_items_in_batch=4).loss
        l8 = model(pixel_values=pixel_values, labels=labels, num_items_in_batch=8).loss
    print(name, "l4/l8 =", (l4 / l8).item(), "(want 2.0)")
```

```
Swin    l4/l8 = 2.0000   (correct: passing a smaller num_items_in_batch scales the loss up)
Swinv2  l4/l8 = 1.0000   (bug: num_items_in_batch is silently ignored)
```

## The fix

Forward `**kwargs` into `self.loss_function(...)` for every affected class,
matching Swin's existing pattern. Every class touched here is registered
under `MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES`, which resolves to
`ForSequenceClassificationLoss` via the `ForImageClassification` → `loss_type`
auto-detection in `PreTrainedModel` — and that loss function *does* use
`num_items_in_batch` (via `fixed_cross_entropy`) when it's given, so the fix
is meaningful, not a no-op, for every one of these.

34 classes across 33 files:

`BeitForImageClassification`, `BitForImageClassification`,
`CLIPForImageClassification`, `ConvNextForImageClassification`,
`ConvNextV2ForImageClassification`, `Data2VecVisionForImageClassification`,
`DinatForImageClassification`, `DonutSwinForImageClassification`,
`EfficientNetForImageClassification`, `FocalNetForImageClassification`,
`HGNetV2ForImageClassification`, `HieraForImageClassification`,
`ImageGPTForImageClassification`, `LevitForImageClassification`,
`MetaClip2ForImageClassification`, `MobileNetV1ForImageClassification`,
`MobileNetV2ForImageClassification`, `MobileViTForImageClassification`,
`MobileViTV2ForImageClassification`, `PerceiverForImageClassificationLearned`,
`PerceiverForImageClassificationFourier`,
`PerceiverForImageClassificationConvProcessing`,
`PoolFormerForImageClassification`, `PvtForImageClassification`,
`PvtV2ForImageClassification`, `RegNetForImageClassification`,
`ResNetForImageClassification`, `SiglipForImageClassification`,
`Siglip2ForImageClassification`, `SwiftFormerForImageClassification`,
`Swinv2ForImageClassification`, `TextNetForImageClassification`,
`TimmWrapperForImageClassification`, `VJEPA2ForVideoClassification`.

Three of these are generated from `modular_*.py` sources
(`HGNetV2ForImageClassification`, `BeitForImageClassification` directly;
`Siglip2ForImageClassification` overrides `forward()` in its own modular
file rather than inheriting it). The fix is applied in the modular source in
those cases, and `modeling_*.py` regenerated via
`utils/modular_model_converter.py` — confirmed consistent with
`utils/check_modular_conversion.py` and `utils/check_copies.py`.

## Test

Added `test_image_classification_loss_respects_num_items_in_batch` to the
shared `ModelTesterMixin` in `tests/test_modeling_common.py`. Rather than
compare loss values (which needs per-architecture tolerances), it wraps
`loss_function` and asserts `num_items_in_batch` actually reaches it — so the
check is exact and architecture-agnostic. It's gated to
`MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES`, so it automatically covers
every current and future model registered there, without needing a
per-model hand-written test.

I verified the test fails against the pre-fix code (reverted `Swinv2` locally
and reran) and passes after the fix, and ran the full non-slow suite for all
33 touched model test files plus this new test: **3405 passed, 0 failed**
(4652 skipped are the usual TF/Flax/slow-marked skips in this environment).

## Scope — what this PR does *not* fix

#47688 identifies ~65 classes across two different bug shapes:

- **Group 1** (`forward()` has `**kwargs`, the loss function *does* use
  `num_items_in_batch`, just needs the kwargs forwarded) — this PR fixes the
  34 image/video classification heads in this group. The issue also flags
  causal-LM/VL heads (e.g. `Qwen3VLForConditionalGeneration`) and text
  sequence/token-classification heads (e.g. `GPTNeoXForSequenceClassification`)
  in the same group — same fix shape, but different call signatures per
  family, so I scoped this PR to the one homogeneous, mechanically-verifiable
  pattern rather than touching ~45 more files with less uniform review risk.
- **Group 2** (DETR-family object detection, RNNT ASR heads) needs the
  opposite fix — `accepts_loss_kwargs = False` — since those loss functions
  don't use item counts at all. Not touched here.

### CODE REVIEW

**Comment by Ifeoluwah** [comment]
cc @molbap @guarin - this is scoped to the image/video classification heads from #47688 (34 classes, shared regression test added). CPU CI is green. Happy to adjust scope if you'd rather see it split differently.

**Comment by Rocketknight1** [comment]
Running a code agent on other people's issues is spamming, not helping.
